swap/swap.go

```diff
@@ -78,9 +80,34 @@ func NewParams() *Params {
 	}
 }
 
+// NewLogger returns a new logger
+func NewLogger(h interface{}) log.Logger {
```

Comment:
This would be first initialized in New Swap.
I agree on making it globally and remove it from the struct itself.

Reply:
fixed

Reply:
I checked the code and it is good now. However, just pointing out that `auditLog` is a global *un-initialized* variable which, if `new()` would not run for some reason, is `nil`, causing panics down the road.

We don't have this situation now I guess, because everything in our package is after a `Swap` object is created.

But, again, if we had a default log file, then this would be non-issue as well as it would be initialized with `newLogger()` right away. :)